projects/storefrontlib/shared/services/file/files-form-validators.ts

```diff
@@ -56,4 +56,30 @@ export class FilesFormValidators {
       return Object.keys(errors).length === 0 ? null : errors;
     };
   }
+
+  /**
+   * Checks allowed types
+   *
+   * @param {Array<string>} allowedTypes Allowed types of files
+   * @returns Uses 'notParsable' validator error with allowedTypes property
+   * @memberOf FilesFormValidators
+   */
+  allowedTypes(allowedTypes?: Array<string>): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
```

Comment:
What if the allowedTypes function is misused? For Example is control is not a fileUpload control, should line 71 be protected or an exception thrown?
Maybe add Unit tests to be sure of the expectations.

Reply:
It won't be an issue as restriction is from backend as well. This is for the better use experience.
And, protected methods can not be accessed from other components.